test/APIBasedAuth.test.ts

```diff
@@ -0,0 +1,116 @@
+import axios from 'axios';
+import APIBasedAuth from '../src/APIBasedAuth';
+
+jest.mock('axios');
+axios.create = jest.fn(() => clientAxios);
+
+let auth: APIBasedAuth, ctorParams;
```

Comment:
Huge nit, I typically like to explicitly declare each variable with separate `let` keyword